packages/serverpod/lib/src/server/session.dart

```diff
@@ -12,10 +13,30 @@ import 'package:serverpod/src/server/serverpod.dart';
 import '../cache/caches.dart';
 import '../database/database.dart';
 
+/// A listener that will be called when the session is about to close.
+typedef WillCloseListener = FutureOr<void> Function(Session);
```

Comment:
I think you can add parameter names to the prototype here, e.g. `Function(Session session)`. This will aid the editor to give a better autocompletions.